lms/djangoapps/certificates/views/xqueue.py

```diff
@@ -48,17 +47,16 @@ def request_certificate(request):
             username = request.user.username
             student = User.objects.get(username=username)
             course_key = CourseKey.from_string(request.POST.get('course_id'))
-            course = modulestore().get_course(course_key, depth=2)
-
             status = certificate_status_for_student(student, course_key)['status']
+
             if can_generate_certificate_task(student, course_key):
                 log.info(f'{course_key} is using V2 course certificates. Attempt will be made to generate a V2 '
                          f'certificate for user {student.id}.')
                 generate_certificate_task(student, course_key)
             elif status in [CertificateStatuses.unavailable, CertificateStatuses.notpassing, CertificateStatuses.error]:
                 log_msg = 'Grading and certification requested for user %s in course %s via /request_certificate call'
                 log.info(log_msg, username, course_key)
-                status = generate_user_certificates(student, course_key, course=course)
```

Comment:
Passing in the course is optional at this point and will be retrieved in the queue.py's `add_cert` function if it is None when generating the certificate.